src/js/program/media-element-pool.ts

```diff
@@ -20,30 +36,31 @@ export default function MediaElementPool() {
     let primed = false;
 
     return {
-        primed() {
+        primed(): boolean {
             return primed;
         },
-        prime() {
+        prime(): void {
             elements.forEach(primeMediaElementForPlayback);
             primed = true;
         },
-        played() {
+        played(): void {
             primed = true;
         },
-        getPrimedElement() {
-            if (pool.length) {
+        getPrimedElement(): HTMLVideoElement | null {
+            const el = pool.shift();
```

Comment:
interestingly TS does not recognize the .length check as a safeguard here - so it's expecting me to cover the case where shift returns a non-element (length 0).

this was my idea of how to fix. based on MDN, this should always return undef for len 0

Reply:
Would it accept `return pool.shift() || null;` ?